hyperspy/model.py

```
@@ -891,6 +898,312 @@ def _model_function(self, param):
+        else:
+            target_signal = self.signal()[np.where(self.channel_switches)]
+            nav_shape = ()
+
+        if is_binned(self.signal):
+            target_signal = target_signal / np.prod(
+                tuple((ax.scale for ax in self.signal.axes_manager.signal_axes))
+            )
+
+        target_signal = target_signal - self._component_data_fixed
+
+        flat_sig_len = target_signal.shape[-1] #np.count_nonzero(self.channel_switches)
+
+        # Reshape what may potentially be Signal2D data into a long Signal1D shape
+        # and an nD navigation shape to a 1D nav shape
+        if self._linear_ndfit:
+            target_signal = target_signal.reshape((np.prod(nav_shape, dtype=int), ) + (flat_sig_len,))
+        else:
+            target_signal = target_signal.reshape((flat_sig_len,))
+
+        if optimizer == "lstsq" and not self.signal._lazy:
+            result, residual, *_ = np.linalg.lstsq(self._component_data.T, target_signal.T, rcond=None)
+            self.coefficient_array = result.T
+
+        elif optimizer == "lstsq" and self.signal._lazy:
+            result, residual, *_ = da.linalg.lstsq(da.asarray(self._component_data).T, target_signal.T)
+            if self._linear_ndfit and (LooseVersion(dask.__version__) < LooseVersion("2020.12.0")):
+                # Dask pre 2020.12 didn't support residuals on 2D input, we calculate them later.
+                residual = None
+            self.coefficient_array = result.T
+
+        elif optimizer == "ridge_regression":
+            if self.signal._lazy:
+                lazy_ridge_warning = (
+                    "You appear to be using the 'ridge_regression' optimizer on a "
+                    "lazy signal. If the signal doesn't fit into memory, it may be much faster to "
+                    "use the 'lstsq' optimizer, as ridge does not support lazy loading with Dask."
+                )
+                warnings.warn(lazy_ridge_warning)
```

If it doesn't support it, then it would raise a error at some point? Either remove this warning and wait for the actual error to come or raise an error here to say that it is not supported.

---

It will eventually cause an error, but it can cause major slowdown of the computer before then. I'm fine with making this raise an error instead of a warning.